lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -0,0 +1,327 @@
+        post[:merchant] = {
+          id: options[:order_id],
+          mcc: options[:merchant_category_codes],
+          country: options[:country]
+        }
+      end
+
+      def add_base_data(post, options)
+        post[:isDeclined] = options[:is_declined]
+        post[:orderId] = options[:order_id]
+        post[:idempotencyKey] = options[:idempotency_key] || options[:order_id]
+      end
+
+      def add_mit_data(post, money, credit_card, options)
+        return unless options[:is_mit]
+
+        post[:isMIT] = options[:is_mit]
+        post[:isRecurring] = options[:is_recurring]
+        post[:expiryDateUtc] = options[:expiry_date_utc]
+
+        post[:isRecurring] && post[:subscription] = {
+          subscriptionId: options[:subscription_id],
+          schemeTransactionId: options[:subscription_scheme_transaction_id],
+          schemeBrand: credit_card.brand,
+          interval: options[:subscription_interval],
+          price: money,
+          currency: options[:currency]
+        }
+      end
+
+      def add_customer_data(post, options)
+        post[:payer] = {
+          id: options[:order_id],
+          email: options[:email] || 'NA',
+          phone: phone_from(options)
+        }
+      end
+
+      def add_address(post, payment, address)
+        first_name, last_name = address_names(address[:name], payment)
+
+        post[:billingInformation] = {
+          firstName: first_name,
+          lastName: last_name,
+          country: address[:country],
+          phone: address[:phone],
+          countryCode: address[:country],
+          addressLine1: address[:address1],
+          state: address[:state],
+          city: address[:city],
+          zipCode: address[:zip]
+        }
+      end
+
+      def add_invoice(post, money, options)
+        post[:transaction] = {
+          id: options[:order_id],
+          dynamicDescriptor: options[:dynamic_descriptor],
+          timestampUtc: options[:timestamp_utc] || Time.now,
+          timezoneUtcOffset: options[:timestamp_utc_offset],
+          amount: money,
+          currency: (options[:currency] || currency(money)),
+          responseCode: options[:response_code],
+          responseCodeSource: options[:response_code_source] || 'NA',
+          avsResultCode: options[:avs_result_code],
+          cvvResultCode: options[:cvv_result_code],
+          cavvResultCode: options[:cavv_result_code] || 'NA',
+          cardNotPresent: options[:card_not_present]
+        }
+      end
+
+      def add_transaction_type(post, action)
+        return unless %i(authorize capture void).include?(action)
+
+        post[:transaction] ||= {}
+        post[:transaction][:transactionType] =
+          case action
+          when :authorize
+            'Auth'
+          when :capture
+            'Capture'
+          when :void
+            'Void'
+          end
+      end
+
+      def add_payment(post, credit_card, address, options)
+        post[:paymentMethod] = {
```

Comment:
Yes, create a flag called `tokenize` to use the tokenize endpoint, when this happens the payment and token fields would be there and used, otherwise the credit card fields per se would be used.